crates/webidl/src/generator.rs

```diff
@@ -695,10 +714,9 @@ impl DictionaryField {
         quote! {
             #unstable_attr
             #cfg_features
-            #doc_comment
-            #unstable_docs
+            #[deprecated = #deprecated]
```

Comment:
I think we should deprecate the old API regardless because it requires `&mut` and having two setter APIs doesn't really make sense.

We could of course make the new setter API use the builder-style pattern though.
I'm personally unconvinced, even though I agree that this API is better, because we would have to generate twice the amount of code for something I think higher level APIs have to wrap up and improve on anyway.

Though I don't have a particularly strong opinion about this, so if you are in favor, I'm happy to just change the new setter API to the builder-style pattern. That would entail making the bindings private again and generating new setters that return `&Self`.

Reply:
> having two setter APIs doesn't really make sense.

I kinda had the opposite thought: I think it's weird how in the current API, the same function is used both for builder-style construction and as a normal setter, because it means that the name has to be unnatural for one of the options.

For example, I think it's less clear what `options.type_("text/javascript")` is doing than `options.set_type("text/javascript")`, and `BlobPropertyBag::new().set_type("text/javascript")` is also a bit off (I guess it's fairly clear what it's doing, but it isn't particularly idiomatic).

But I don't have a particularly strong opinion either, and I can see the benefit of keeping it simple.

Reply:
Good point actually.

But the reason this usually doesn't work is because idiomatically getters don't use get prefixed with `get_`, so this would collide with the builder-style setters.

So the way I see it we can either:
1. Not deprecate the builder-style setters and at the next breaking change just remove their `&mut` requirement.
2. We move on with deprecating the builder-style setters and at the next breaking change remove them and rename getters idiomatically from `get_field()` to `field()`.

I'm still tending towards 2., but I don't have a particularly strong opinion as well.
Let me know if you want me to go with 1. instead and I can change that, otherwise I'm happy to proceed with how it is right now.